index.js

```diff
@@ -1,60 +1,68 @@
 'use strict'
 
-const pLimit = require('p-limit')
+async function * mapIterator (iterator, func, n = 16) {
+  const promises = []
```

Comment:
This array will grow indefinitely, it should remove elements after they are resolved, keeping memory usage costant.

Reply:
It is being removed, see `promies.shift()`

Reply:
A few inline comment explaning the two parts that are run in parallel would be amazing.